### PR TITLE
Fail actors if state does not have trx and etag + improve transaction…

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Fail actors if state does not have trx and etag + improve transactional store loading.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2890 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [X] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/http/api_test.go [hunk 1]
```diff
@@ -2612,15 +2623,19 @@ func TestV1HealthzEndpoint(t *testing.T) {
 
 func TestV1TransactionEndpoints(t *testing.T) {
 	fakeServer := newFakeHTTPServer()
-	fakeStore := fakeStateStore{}
+	var fakeStore state.Store = fakeStateStore{}
```
Comment (pkedy) [inline pkg/http/api_test.go:2626]: same

File: pkg/http/api_test.go [hunk 2]
```diff
@@ -2094,13 +2094,17 @@ func (f *fakeHTTPServer) DoRequest(method, path string, body []byte, params map[
 func TestV1StateEndpoints(t *testing.T) {
 	etag := "`~!@#$%^&*()_+-={}[]|\\:\";'<>?,./'"
 	fakeServer := newFakeHTTPServer()
-	fakeStore := fakeStateStore{}
+	var fakeStore state.Store = fakeStateStore{}
```
Comment (pkedy) [inline pkg/http/api_test.go:2097]: nit: You can let the `fakeStore` variable remain type `fakeStateStore` and let Go cast it for you in `fakeTransactionalStores`. Sort of along the lines of "Return implementations and accept interfaces".

**Review by pkedy — APPROVED** [review]
LGTM. super microscopic nit

File: pkg/http/api_test.go (hunk 1 above)
Comment (artursouza) [reply pkg/http/api_test.go:2626]: same

File: pkg/http/api_test.go (hunk 2 above)
Comment (artursouza) [reply pkg/http/api_test.go:2097]: It doesn't. I get a compilation error. Compiler wins!

pkg/http/api_test.go:2102:22: invalid type assertion: fakeStore.("github.com/dapr/components-contrib/state".TransactionalStore) (non-interface type fakeStateStore on left)

Comment (pkedy) [reply pkg/http/api_test.go:2097]: For future reference (no need to type assertions)
```golang
var fakeStore fakeStateStore
fakeStores := map[string]state.Store{
	"store1": fakeStore,
}
fakeTransactionalStores := map[string]state.TransactionalStore{
	"store1": fakeStore,
}
```

**Pull request merged**